developers/utils/i18n.md

```diff
@@ -444,6 +444,17 @@ The custom keys are a very good practice to translate bundle dependent error mes
 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "@text/offline.communication-error");
 ```
 
+## Generating I18N properties file
+
+To automatically generate an I18N properties file from XML files, the following command can be run:
+
+```shell
+mvn org.openhab.core.tools:i18n-maven-plugin:3.2.0:generate-default-translations
```

Comment:
Does the `3.2.0` need to be adjusted everytime a new OH version is released? I read in the README.md that the simpler command might not always work. Do you know why that is and when? Maybe we should use the simpler command here, nonetheless to keep the readme free from version numbers.

Reply:
@wborn - can you explain this? I've also been wondering about this, and I haven't heard from anyone who was able to simply run `mvn i18n:generate-default-translations` as mentioned in the [openHAB 3.2 release announcement](https://www.openhab.org/blog/2021-12-19-openhab-3-2-release.html). Also, currently towards main (3.3), I cannot run `mvn org.openhab.core.tools:i18n-maven-plugin:3.3.0:generate-default-translations` either.

Reply:
>  can you explain this?

I think I've seen it work but then again it no longer does. :thinking:

Maybe something is missing, these links explain how plugin prefixes work:

* https://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html
* https://books.sonatype.com/mvnref-book/reference/writing-plugins-sect-custom-plugin.html#writing-plugins-sect-plugin-prefix

> I cannot run mvn org.openhab.core.tools:i18n-maven-plugin:3.3.0:generate-default-translations either.

That will only work when `3.3.0` is released... until that time you should use `3.3.0-SNAPSHOT`.

Reply:
@wborn - I tried "3.3.0-SNAPSHOT" also before posting, but didn't work for me:
`[WARNING] The POM for org.openhab.core.tools:i18n-maven-plugin:jar:3.3.0-SNAPSHOT is missing, no dependency information available`

Reply:
This should work:
`mvn org.openhab.core.tools:i18n-maven-plugin:3.3.0-SNAPSHOT:generate-default-translations`

Reply:
I get this result:
```
$ mvn org.openhab.core.tools:i18n-maven-plugin:3.3.0-SNAPSHOT:generate-default-translations
[INFO] Scanning for projects...
Downloading from central: https://repo1.maven.org/maven2/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/maven-metadata.xml
Downloading from openhab-release: https://openhab.jfrog.io/openhab/libs-release/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/maven-metadata.xml
Downloading from central: https://repo1.maven.org/maven2/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/i18n-maven-plugin-3.3.0-SNAPSHOT.pom
Downloading from openhab-release: https://openhab.jfrog.io/openhab/libs-release/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/i18n-maven-plugin-3.3.0-SNAPSHOT.pom
[WARNING] The POM for org.openhab.core.tools:i18n-maven-plugin:jar:3.3.0-SNAPSHOT is missing, no dependency information available
Downloading from central: https://repo1.maven.org/maven2/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/i18n-maven-plugin-3.3.0-SNAPSHOT.jar
Downloading from openhab-release: https://openhab.jfrog.io/openhab/libs-release/org/openhab/core/tools/i18n-maven-plugin/3.3.0-SNAPSHOT/i18n-maven-plugin-3.3.0-SNAPSHOT.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.956 s
[INFO] Finished at: 2022-06-02T23:08:45+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.openhab.core.tools:i18n-maven-plugin:3.3.0-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact org.openhab.core.tools:i18n-maven-plugin:jar:3.3.0-SNAPSHOT in central (https://repo1.maven.org/maven2)
```

Reply:
Currently you'll first have to build snapshot versions of the plugin locally for that command to work.
That's because there is no snapshot Maven repo for plugins defined in the openHAB super POM.
I've created https://github.com/openhab/infrastructure/pull/14 to also make it possible to download snapshot versions of openHAB Maven plugins.

Reply:
Going back to original comment: @fwolter - it seems like the version number is needed, unfortunately, at least for the time being. @Confectrician, is it possible to dynamically include version number somehow?

Reply:
Should be possible in theory.

There are 3 possible options:

- Vuepress knows the version already for the builds as global variable
- We introduce a variable via frontmatter of the markdown page, which we then write during the buil
- We search and replace a placeholder in our ruby scripts here and in the website repository.

@ghys 
Maybe you have something in mind without the need of checking the docs?

Reply:
Now that there is a new openhab-super-pom (https://github.com/openhab/infrastructure/issues/15), `mvn i18n:generate-default-translations` should also work. :slightly_smiling_face:

Reply:
Thanks, `mvn org.openhab.core.tools:i18n-maven-plugin:generate-default-translations` is now working for me. I can also run `mvn i18n:generate-default-translations`, but it generates this warning twice:

```
[WARNING] The POM for org.eclipse.m2e:lifecycle-mapping:jar:1.0.0 is missing, no dependency information available
[WARNING] Failed to retrieve plugin descriptor for org.eclipse.m2e:lifecycle-mapping:1.0.0: Plugin org.eclipse.m2e:lifecycle-mapping:1.0.0 or one of its dependencies could not be resolved: org.eclipse.m2e:lifecycle-mapping:jar:1.0.0 was not found in https://openhab.jfrog.io/openhab/libs-snapshot during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of openhab-snapshot has elapsed or updates are forced
```

I tried also with option -U, but still getting these warnings.

Reply:
Maybe you use an older Maven version where it does not work properly @jlaur?

It works fine with Maven 3.8.6 on macOS/Linux/Windows with both Java 11/17, see this matrix build: :slightly_smiling_face: 

https://github.com/wborn/openhab-addons/runs/7262496169?check_suite_focus=true

Reply:
The m2e warnings are unrelated, perhaps we can prevent them by moving it to an Eclipse only profile like [this](https://stackoverflow.com/a/23707050).

Reply:
I've created a few PRs to address these warnings:

* https://github.com/openhab/openhab-core/pull/3034
* https://github.com/openhab/openhab-addons/pull/13098
* https://github.com/openhab/openhab-webui/pull/1446

Reply:
> It works fine with Maven 3.8.6 on macOS/Linux/Windows with both Java 11/17, see this matrix build: 🙂

Thanks for looking into this, I'll verify when home from holiday. 🙂 And create PR for simplifying docs.

Reply:
#1867 created.